python/github_repo_cloner.py

```diff
@@ -3,12 +3,24 @@
 import requests
 
 from git import Repo
-username = raw_input("Please enter your Github account name: ")
-response = requests.get("https://api.github.com/users/" + username + "/repos")
+username = input("Please enter your Github account name: ")
+url = "https://api.github.com/users/" + username + "/repos?per_page=30"
+response = requests.get(url)
 repositories =  response.json()
+try:
+    current_repos = os.listdir("./my_repositories/")
+except FileNotFoundError:
+    current_repos = []
 
-for repo in repositories:
-	command = git.Git().clone(repo["clone_url"], "my_repositories/" + repo["name"], recursive=True)
-	r = Repo("my_repositories/" + repo["name"])
-	r.create_remote('originate', repo["clone_url"])
-	print command
\ No newline at end of file
+page = 1
+while len(repositories) > 0:
+    print('page ' + str(page))
+    page += 1
+    for repo in repositories:
+        if repo["name"] not in current_repos:
+            command = git.Git().clone(repo["clone_url"], "my_repositories/" + repo["name"], recursive=True)
+            r = Repo("my_repositories/" + repo["name"])
+            r.create_remote('originate', repo["clone_url"])
+            print(repo["name"])
+    new_response = requests.get(url + "&page=" + str(page))
+    repositories = new_response.json()
```

Comment:
Kemptosa says, this will allow the python implementation to get closer to having the same functionality as the ruby version. Guess that the ruby version is superior at the moment.

thanks @kemptosa , great to have some python expertise on the team. I will review later.

Reply:
@MichaelDimmitt This is super old code, but if I recall correctly the Python Github library had more abilities. It looks like we ended up doing most of the Ruby implementation with straight HTTP calls to Github.